recipes/cad1/task2/baseline/car_scene_acoustics.py

```diff
@@ -0,0 +1,260 @@
+        self.hrir_values["p90_left"] = wavfile.read(
+            anechoic_hrtf_dir / "HR72_LSP01_CH1_Left.wav"
+        )[1]
+        self.hrir_values["p90_right"] = wavfile.read(
+            anechoic_hrtf_dir / "HR72_LSP01_CH1_Right.wav"
+        )[1]
+
+    def apply_hearing_aid(
+        self, signal: np.ndarray, audiogram: np.ndarray, center_frequencies: np.ndarray
+    ) -> np.ndarray:
+        """
+        Applies the hearing aid:
+        It consists in NALR prescription and Compressor
+
+        Args:
+            signal (np.ndarray): The audio signal to be enhanced.
+            audiogram (np.ndarray): An audiogram used to configure the NALR object.
+            center_frequencies (np.ndarray): An array of center frequencies
+                used to configure the NALR object.
+
+        Returns:
+            np.ndarray: The enhanced audio signal.
+        """
+        nalr_fir, _ = self.enhancer.build(audiogram, center_frequencies)
+        signal = self.enhancer.apply(nalr_fir, signal)
+        signal, _, _ = self.compressor.process(signal)
+        return signal
+
+    def add_anechoic_hrtf(self, noise_signal: np.ndarray) -> np.ndarray:
+        """
+        Adds the Anechoic HRTF to the noise signal.
+        Args:
+            noise_signal: A numpy array representing the different components
+                of the car noise signal.
+
+        Returns:
+            np.ndarray: The noise signal with the Anechoic HRTF applied.
+
+        """
+        # Apply Anechoic HRTF to the noise signal
+        # Engine first
+        out_left = lfilter(self.hrir_values["000_left"], 1, noise_signal[0, :])
+        our_right = lfilter(self.hrir_values["000_right"], 1, noise_signal[0, :])
+
+        # noise processing hardwired for 2 noises
+        out_left += lfilter(self.hrir_values["m90_left"], 1, noise_signal[1, :])
+        our_right += lfilter(self.hrir_values["m90_right"], 1, noise_signal[1, :])
+
+        # swap HRIR so this noise is on the other side
+        out_left += lfilter(self.hrir_values["p90_left"], 1, noise_signal[2, :])
+        our_right += lfilter(self.hrir_values["p90_right"], 1, noise_signal[2, :])
+
+        noise_anechoic = np.stack([out_left, our_right], axis=0)
+
+        return noise_anechoic
+
+    def get_car_noise(
+        self,
+        car_noise_params: dict,
+    ) -> np.ndarray:
+        """
+        Generates car noise.
+
+        Args:
+            car_noise_params (dict): Car Noise Parameters as generated by
+                Class CarNoiseParameterGenerator
+
+        Returns:
+            numpy.ndarray: A numpy array representing the different components
+                of the car noise signal
+
+        """
+        noise_signal = self.carnoise.generate_car_noise(
+            noise_parameters=car_noise_params,
+            number_noise_sources=2,
+            commonness_factor=0,
+        )
+        return noise_signal
+
+    def add_car_hrtf(self, signal: np.ndarray, hrir: dict) -> np.ndarray:
```

Comment:
I'm of the mindset that we should do things right the first time as typically we don't return to code a refactor it until the technical debt has accumulated to the point where its hindering development.

I guess since this code is isolated for the specific Cadenza challenge it won't have too much of an impact but should it form the basis of a subsequent Cadenza challenge then it should be correctly refactored into `pyclarity` so that its reused.

Reply:
No, it is very unlikely we will reuse this car scenario in subsequent challenges

Reply:
I also try to do everything in the challenges so we don't need to revise them. But, in this case, I am not sure how to set both HRTFs in a single method.
The only way I can see how to join these 2 methods is by using a dictionary that maps signals with HRTFs. But, this approach seems very confusing to use.